migrations/Version20180810134011.php

```diff
@@ -0,0 +1,163 @@
+  private static $JARS_BOX_NAME = "jars";
+  private static $PASSED_JARS_BOX_NAME = "jars-passed";
+  private static $FILES_IN_TYPE = "files-in";
+  private static $FILES_OUT_TYPE = "files-out";
+  private static $REMOTE_FILES_TYPE = "remote-file[]";
+
+  /**
+   * @return array java compilation changed pipelines
+   * @throws DBALException
+   */
+  private function updatePipelines(): array {
+    $changedPipelines = [];
+    $pipelinesResult = $this->connection->executeQuery("SELECT * FROM pipeline");
+    foreach ($pipelinesResult as $pipeline) {
+      $pipelineConfig = $this->connection->executeQuery("SELECT * FROM pipeline_config WHERE id = '{$pipeline["pipeline_config_id"]}'")->fetch();
+      $config = Yaml::parse($pipelineConfig["pipeline_config"]);
+      $changed = false;
+
+      foreach ($config["boxes"] as &$box) {
+        if ($box["type"] === self::$JAVAC_BOX) {
+          $changed = true;
+          $changedPipelines[] = $pipeline["id"];
+
+          $box["portsIn"][self::$JAR_FILES_PORT] = [
+            "type" => self::$FILES_TYPE,
+            "value" => self::$JAR_FILES_PORT
+          ];
+          // input jar files in compilation
+          $config["boxes"][] = [
+            "name" => self::$JARS_BOX_NAME,
+            "type" => self::$FILES_IN_TYPE,
+            "portsIn" => [],
+            "portsOut" => [
+              "input" => [
+                "type" => self::$FILES_TYPE,
+                "value" => self::$JAR_FILES_PORT
+              ]
+            ]
+          ];
+          // output jar files from compilation
+          $config["boxes"][] = [
+            "name" => self::$PASSED_JARS_BOX_NAME,
+            "type" => self::$FILES_OUT_TYPE,
+            "portsIn" => [
+              "output" => [
+                "type" => self::$FILES_TYPE,
+                "value" => self::$JAR_FILES_PORT
+              ]
+            ],
+            "portsOut" => []
+          ];
+          $config["variables"][] = [
+            "name" => self::$JAR_FILES_PORT,
+            "type" => self::$FILES_TYPE,
+            "value" => []
+          ];
+        }
+
+        if ($box["type"] === self::$JAVA_RUNNER_BOX) {
+          $changed = true;
+
+          $box["portsIn"][self::$JAR_FILES_PORT] = [
+            "type" => self::$FILES_TYPE,
+            "value" => self::$JAR_FILES_PORT
+          ];
+          // input jar files in execution
+          $config["boxes"][] = [
+            "name" => self::$JARS_BOX_NAME,
+            "type" => self::$FILES_IN_TYPE,
+            "portsIn" => [],
+            "portsOut" => [
+              "input" => [
+                "type" => self::$FILES_TYPE,
+                "value" => self::$JAR_FILES_PORT
+              ]
+            ]
+          ];
+          $config["variables"][] = [
+            "name" => self::$JAR_FILES_PORT,
+            "type" => self::$FILES_TYPE,
+            "value" => []
+          ];
+        }
+      }
+
+      if ($changed) {
```

Comment:
This flag is redundant. You can get the same bit of information as !empty($changedPipelines); (which is even more comprehensible IMHO).

Reply:
Noted... this is a relict from previous implementation which was changed.

Reply:
Well looking on the implementation... the `changed` variable is actually needed here. There are two `if` branches, the `changedPipelines` is changed only in one of them.